rc/src/lib.rs

```diff
@@ -3,13 +3,21 @@ use std::ops::Deref;
 pub struct MyRc<T> {
     ptr: *mut T,
     refcount: *mut usize,
+    weakcount: *mut usize,
 }

```

Comment:
This layout definitely works, but introduces some extra indirection (you have to follow two pointer to get to the `T`). If you think about it hard enough, you can actually avoid the extra indirection.